lib/tts_plugin.dart

```diff
@@ -52,29 +55,49 @@ class TTSQueue {
   Future<void> speak(String id, String text) async {
     final completer = Completer<void>();
     final element = (id: id, text: text, completer: completer);
-    if (queue.isNotEmpty) {
-      final previous = queue.last;
-      queue.addLast(element);
-      await previous.completer.future;
-      if (queue.firstOrNull != element) {
-        throw Exception('Message was deleted');
+
+    // Logic to manage TTS and username reading based on queue size
+    manageTTSState();
+
+    if (isTTSDisabled) {
+      if (queue.length == 1) {
+        // Only read the TTS disabled message once
+        await TextToSpeechPlugin.speak(
+            "There are too many messages. TTS Disabled");
       }
-      await TextToSpeechPlugin.speak(text);
       completer.complete();
     } else {
-      queue.addLast(element);
-      await TextToSpeechPlugin.speak(text);
-      completer.complete();
+      if (queue.isNotEmpty) {
+        final previous = queue.last;
+        queue.addLast(element);
+        await previous.completer.future;
+        if (queue.firstOrNull != element) {
+          throw Exception('Message was deleted');
+        }
+        final message = isUsernameReadingEnabled ? text : removeUsername(text);
+        await TextToSpeechPlugin.speak(message);
+        completer.complete();
+      } else {
+        queue.addLast(element);
+        final message = isUsernameReadingEnabled ? text : removeUsername(text);
+        await TextToSpeechPlugin.speak(message);
+        completer.complete();
+      }
     }
+
     queue.remove(element);
+    manageTTSState(); // Re-check the state in case the queue is now empty
   }
 
   void delete(String id) {
     queue.removeWhere((speak) => speak.id == id);
+    manageTTSState();
   }
 
   Future<void> clear() async {
     queue.clear();
+    isUsernameReadingEnabled = true;
+    isTTSDisabled = false;
```

Comment:
If tts is disabled, we will need to send a signal to the other thread as this will not show the tts getting disabled on the button.

Reply:
Ok I think I got it to the other thread.